homeworks/src/kpobozhny/homework/hw_04/triangle/Triangle.java

```diff
@@ -0,0 +1,114 @@
+package kpobozhny.homework.hw_04.triangle;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+/**
+ * Created by kostya on 4/22/17.
+ */
+
+/*        Создайте класс треугольников на координатной плоскости,
+        используя в качестве полей объекты-точки.
+        Реализуйте в классе:
+        а) конструктор, позволяющий задавать вершины;
+        b) метод print(), выводящий описание треугольника на экран;
+        c) методы для вычисления периметра и площади треугольника.
+        Инкапсулируйте поля таким образом, чтобы нельзя было изменить значение любого из них так,
+        чтобы вершины оказались на одной прямой.*/
+
+public class Triangle {
+
+    private Point A, B, C;
+
+    public Triangle(Point a, Point b, Point c) {
+        this.A = a;
+        this.B = b;
+        this.C = c;
+    }
+
+    public Point getVertexA() {
+        return A;
+    }
+
+    public void setVertexA(Point point) {
+        if (((B.getX() == point.getX()) && (C.getX() == point.getX())) ||
+                ((B.getY() == point.getY()) && (C.getY() == point.getY()))) {
+            System.out.println("Sorry, your coordinates are invalid (All triangle's vertices can not belong to the same line)!");
+        } else {
+            this.A = point;
+        }
+
+    }
+
+    public Point getVertexB() {
+        return B;
+    }
+
+    public void setVertexB(Point point) {
+        if (((A.getX() == point.getX()) && (C.getX() == point.getX())) ||
+                ((A.getY() == point.getY()) && (C.getY() == point.getY()))) {
+            System.out.println("Sorry, your coordinates are invalid (All triangle's vertices can not belong to the same line)!");
+        } else {
+            this.B = point;
+        }
+
+    }
+
+    public Point getVertexC() {
+        return C;
+    }
+
+    public void setVertexC(Point point) {
+        if (((A.getX() == point.getX()) && (B.getX() == point.getX())) ||
+                ((A.getY() == point.getY()) && (B.getY() == point.getY()))) {
```

Comment:
Code duplication - you can avoid it if you will store your vertexes in array for example. (Not A,B,C but 1,2,3 and method signature would be not setVertexC(point) but setVertex(index, point))